google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/standard/AppEngineStandardFacet.java

```diff
@@ -85,10 +94,22 @@ public boolean isNonStandardCompatEnvironment() {
     return isFlexCompatEnvironment() || isManagedVmCompatEnvironment();
   }
 
-  public boolean isJava8Runtime() {
+  /**
+   * Returns the {@link JavaVersion} of the Java runtime, as defined in the {@code
+   * appengine-web.xml}.
+   *
+   * @return an {@link Optional} of the runtime {@link JavaVersion}. Returns {@link
+   *     Optional#empty()} if the {@code appengine-web.xml} does not exist or if the value of the
+   *     runtime is not recognized.
+   */
+  public Optional<JavaVersion> getRuntimeJavaVersion() {
     AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
-    return appEngineStandardWebApp != null
-        && "java8".equalsIgnoreCase(appEngineStandardWebApp.getRuntime().getStringValue());
+    if (appEngineStandardWebApp == null) {
+      return Optional.empty();
+    }
+
+    String runtime = appEngineStandardWebApp.getRuntime().getStringValue();
+    return Optional.ofNullable(RUNTIMES_MAP.get(runtime));
```

Comment:
shouldn't null here resolve to java7?

Reply:
@etanshaul discussed this yesterday and decided that we would return an `Optional.empty()` for not specified runtimes and unrecognized runtimes here; the caller can decide to interpret this as Java7 if they want. But I'm open to suggestions, I have mixed feelings about both approaches.